### PR TITLE
fix(aws-iam): fixes #2041 sensitive module, support required to replace current cfn template construction

### LINKED ISSUE #2041: PolicyPrincipal assumeRoleAction should support multiple actions
currently, assumeRoleAction for PolicyPrincipals in the iam package doesn't support multiple actions and is defined as a string.  Some orgs require assumeRole and assumeRoleWithSAML and this currently isn't possible.  This type should support ```string | Array<string>```

### PR DESCRIPTION
**Pull request by reillykw**
Support added to PolicyDocument assume role actions which is needed for policy documents that require more than one action

----

### Pull Request Checklist

* [ ] Testing
  - Unit test added (prefer not to modify an existing test, otherwise, it's probably a breaking change)
  - __CLI change?:__ coordinate update of integration tests with team
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [ ] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
* [ ] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [x] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: packages/@aws-cdk/aws-iam/lib/policy-document.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ export abstract class PolicyPrincipal {
   /**
    * When this Principal is used in an AssumeRole policy, the action to use.
    */
-  public assumeRoleAction: string = 'sts:AssumeRole';
+  public assumeRoleAction: string | Array<string> = 'sts:AssumeRole';
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-iam/lib/policy-document.ts]: We don't expose type unions at the L2 level, so we can't accept this.

**Comment by rix0rrr** [comment]
I think instead of multiple assume role actions, making it so that `CompositePrincipal` can render to multiple policy statements would serve the same purpose, and be more generally useful in solving:

https://github.com/awslabs/aws-cdk/issues/1578

Are you interested in taking that on?

**Comment by reillykw** [comment]
Yeah I could give it a shot.

**Comment by eladb** [comment]
@reillykw any updates on this?

**Comment by rix0rrr** [comment]
I am going to close this PR as it does not look there's going to be any movement on it. Feel free to reopen in case of renewed interest.

File: packages/@aws-cdk/aws-iam/lib/policy-document.ts (hunk 1 above)
Comment (reillykw) [reply packages/@aws-cdk/aws-iam/lib/policy-document.ts]: How might this be achieved otherwise?  Any suggestions would be helpful.

**Pull request closed without merging**